drivers/rtc/rtc_rv8263.c

```diff
@@ -0,0 +1,782 @@
+		err = i2c_reg_write_byte_dt(&config->i2c_bus, RV8263C8_REGISTER_MINUTES_ALARM,
+					    RV8263C8_BM_ALARM_ENABLE | regs[1]);
+	} else {
+		err = i2c_reg_write_byte_dt(&config->i2c_bus, RV8263C8_REGISTER_MINUTES_ALARM,
+					    RV8263C8_BM_ALARM_DISABLE);
+	}
+
+	if (err < 0) {
+		LOG_ERR("Error while writing MINUTE alarm! Error: %i", err);
+		return err;
+	}
+
+	if (mask & RTC_ALARM_TIME_MASK_HOUR) {
+		err = i2c_reg_write_byte_dt(&config->i2c_bus, RV8263C8_REGISTER_HOURS_ALARM,
+					    RV8263C8_BM_ALARM_ENABLE | regs[2]);
+	} else {
+		err = i2c_reg_write_byte_dt(&config->i2c_bus, RV8263C8_REGISTER_HOURS_ALARM,
+					    RV8263C8_BM_ALARM_DISABLE);
+	}
+
+	if (err < 0) {
+		LOG_ERR("Error while writing HOUR alarm! Error: %i", err);
+		return err;
+	}
+
+	if (mask & RTC_ALARM_TIME_MASK_MONTHDAY) {
+		err = i2c_reg_write_byte_dt(&config->i2c_bus, RV8263C8_REGISTER_DATE_ALARM,
+					    RV8263C8_BM_ALARM_ENABLE | regs[3]);
+	} else {
+		err = i2c_reg_write_byte_dt(&config->i2c_bus, RV8263C8_REGISTER_DATE_ALARM,
+					    RV8263C8_BM_ALARM_DISABLE);
+	}
+
+	if (err < 0) {
+		LOG_ERR("Error while writing MONTHDAY alarm! Error: %i", err);
+		return err;
+	}
+
+	if (mask & RTC_ALARM_TIME_MASK_WEEKDAY) {
+		err = i2c_reg_write_byte_dt(&config->i2c_bus, RV8263C8_REGISTER_WEEKDAY_ALARM,
+					    RV8263C8_BM_ALARM_ENABLE | regs[4]);
+	} else {
+		err = i2c_reg_write_byte_dt(&config->i2c_bus, RV8263C8_REGISTER_WEEKDAY_ALARM,
+					    RV8263C8_BM_ALARM_DISABLE);
+	}
+
+	if (err < 0) {
+		LOG_ERR("Error while writing WEEKDAY alarm! Error: %i", err);
+	}
+
+	return err;
+}
+
+static int rv8263c8_alarm_get_time(const struct device *dev, uint16_t id, uint16_t *p_mask,
+				   struct rtc_time *timeptr)
+{
+	int err;
+	const struct rv8263c8_config *config = dev->config;
+	uint8_t value[5];
+
+	ARG_UNUSED(id);
+
+	if (timeptr == NULL) {
+		return -EINVAL;
+	}
+
+	(*p_mask) = 0;
+
+	err = i2c_burst_read_dt(&config->i2c_bus, RV8263C8_REGISTER_SECONDS_ALARM, value,
+				sizeof(value));
+	if (err < 0) {
+		LOG_ERR("Error while reading alarm! Error: %i", err);
+		return err;
+	}
+
+	if (value[0] <= MAX_SEC) {
+		timeptr->tm_sec = bcd2bin(value[0]) & SECONDS_BITS;
+		(*p_mask) |= RTC_ALARM_TIME_MASK_SECOND;
+	}
+
+	if (value[1] <= MAX_MIN) {
+		timeptr->tm_min = bcd2bin(value[1]) & MINUTES_BITS;
+		(*p_mask) |= RTC_ALARM_TIME_MASK_MINUTE;
+	}
+
+	if (value[2] <= MAX_HOUR) {
+		timeptr->tm_hour = bcd2bin(value[2]) & HOURS_BITS;
+		(*p_mask) |= RTC_ALARM_TIME_MASK_HOUR;
+	}
+
+	if (value[3] <= MAX_MDAY) {
+		timeptr->tm_mday = bcd2bin(value[3]) & DATE_BITS;
+		(*p_mask) |= RTC_ALARM_TIME_MASK_MONTHDAY;
+	}
+
+	if (value[4] <= MAX_WDAY) {
+		timeptr->tm_wday = bcd2bin(value[4]) & WEEKDAY_BITS;
+		(*p_mask) |= RTC_ALARM_TIME_MASK_WEEKDAY;
+	}
+
+	return 0;
+}
+
+static int rv8263c8_alarm_set_callback(const struct device *dev, uint16_t id,
+				       rtc_alarm_callback callback, void *user_data)
+{
+	int err;
+	struct rv8263c8_data *const data = dev->data;
```

Comment:
the pointer does not need to be const, it has no negative impact, but it is not usually done in-tree :)